e2e/util/placement.go

```diff
@@ -103,10 +114,20 @@ func getPlacementDecisionFromPlacement(ctx types.Context, placement *v1beta1.Pla
 	plDecision := plDecisions.Items[0]
 	// r.Log.Info("Found ClusterDecision", "ClsDedicision", plDecision.Status.Decisions)
 
-	if len(plDecision.Status.Decisions) > 1 {
+	decisionCount := 0
+
+	for idx := range plDecision.Status.Decisions {
+		if plDecision.Status.Decisions[idx].Reason == PlacementDecisionReasonFailoverRetained {
+			continue
+		}
+
+		decisionCount++
+	}
```

Comment:
This function has only one caller: getClusterDecisionFromPlacement. Since this caller is interested only in the cluster decision, we can do the filtering here once, fail if we found more than one decision, and return the single cluster decision. This will save repeating the filtering in the caller.